File: solutions/_1624_largest_substring_between_two_equal_characters/solution.cpp
#include <string>
#include <map>
using namespace std;

class Solution
{
public:
    int maxLengthBetweenEqualCharacters(string s)
    {
        map<char, int> first;
        int ans = -1;
        for (int i = 0; i < s.size(); i++)
        {
            if (first.find(s[i]) == first.end())
            {
                first[s[i]] = i;
            }
            else
            {
                ans = max(ans, i - first[s[i]] - 1);
            }
        }
        return ans;
    };
};

/*
https://leetcode.com/problems/largest-substring-between-two-equal-characters/
*/

File: solutions/_1624_largest_substring_between_two_equal_characters/solution.js
/**
 * @param {string} s
 * @return {number}
 */
var maxLengthBetweenEqualCharacters = function (s) {
  const charToIndex = new Map();
  let max = -1;
  for (let i = 0; i < s.length; i++) {
    if (charToIndex.has(s[i])) {
      max = Math.max(max, i - charToIndex.get(s[i]) - 1);
    } else {
      charToIndex.set(s[i], i);
    }
  }
  return max;
};

module.exports = { maxLengthBetweenEqualCharacters };

/*
https://leetcode.com/problems/largest-substring-between-two-equal-characters/
*/
